### ISSUE TITLE
Support for Samsung Pass

### ISSUE THREAD

**Issue by yazantahhan** [NiklasMerz/cordova-plugin-fingerprint-aio#11]
Hi,

I have an issue with some devices that the plugin can't detect the Fingerprint functionality even it is supported and usable.

I tested the plugin and detected the Fingerprint functionality using OnePlus 2 and Samsung Galaxy S7 edge devices.

However, it failed with Samsung Galaxy Note 4 and Samsung Galaxy S5. It gives me the following error in the error callback:

>  Fingerprint authentication not ready

 

My project is developed using Ionic 1 and Typescript. I installed the latest version of the plugin.

Here is the code:

`Fingerprint.isAvailable((result) => {
    console.log('true');
 }, (result) => {
    console.log(result);
});`

Thanks.

**Comment by NiklasMerz** [comment]
Could you please give me more informationabout those Samsung devices? Are they running Android 6 or above?

**Comment by yzn002** [comment]
Thank you for the reply. Both have Android 6.0.1

**Comment by NiklasMerz** [comment]
I am not sure but maybe you @yzn002 could find out. Those devices possibly use Samsungs own fingerprint authentication software / API. This Cordova plugin currently only supports the native Android fingerprint [API](https://developer.android.com/about/versions/marshmallow/android-6.0.html). Some Samsung devices (probably those that came out prior the Android 6.0 release) use another their own interface, as documented [here](http://developer.samsung.com/galaxy/pass).

**Comment by yzn002** [comment]
Nice catch! Actually I got this issue with some galaxy device users and reported the problem, so neither I have a Galaxy device (I have a OnePlus 2 device which works as supposed). But I will try to figure it out

**Comment by walter-ghost** [comment]
not only samsung phone's problem also LG same case. Maybe it's android version problem. 7.0 version phones does not support this plugin

**Comment by NiklasMerz** [comment]
Thanks for reporting. I have never heard of a custom fingerprint API by LG. Samsung uses Samsung Pass on some devices.

**Comment by RaedShari** [comment]
not working here, for Galaxy note 4.

**Comment by samastur** [comment]
Does anyone have experience with Huawei?

I am getting "not ready" error on P10 with Android 7.0 and EMUI 5.1 overlay.

**Comment by tarunaahuja20** [comment]
Not working for Moto G too. Any help please?

**Comment by NiklasMerz** [comment]
Which Android version is your Moto G running? Does it use the standard Android fingerprint API?

**Comment by dstefanoska** [comment]
I'm getting this error "Fingerprint authentication not ready" on Nexus 6P running Android 8.0.1 and also on Infinix phone running Andorid GO. However this error is only happening when I haven't set any fingerprint lock on my phone (they have the fingerprint hardware though). I searched a bit and tried to debug. I also tried to inject the plugin in the promise for platform ready but I still get the error. Any news on why this is happening?

I'm using Ionic 4, fingerprint-aio version ~1.6.0 and @ionic-native/fingerprint-aio version ^4.20.0
